packages/updating-element/src/updating-element.ts

```diff
@@ -926,3 +1024,50 @@ export abstract class UpdatingElement extends HTMLElement {
 // Apply polyfills if available
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 (globalThis as any)['updatingElementPlatformSupport']?.({UpdatingElement});
+
+// Dev mode warnings...
+if (DEV_MODE) {
+  const requestUpdateThenable = {
+    then: (
+      onfulfilled?: (value: boolean) => void,
+      _onrejected?: () => void
+    ) => {
+      console.warn(
+        `\`requestUpdate\` no longer returns a Promise.` +
+          `Use \`updateComplete\` instead.`
+      );
+      if (onfulfilled !== undefined) {
+        onfulfilled(false);
+      }
+    },
+  };
+
+  const requestUpdate = UpdatingElement.prototype.requestUpdate;
+  UpdatingElement.prototype.requestUpdate = function (
+    this: UpdatingElement,
+    name?: PropertyKey,
+    oldValue?: unknown,
+    options?: PropertyDeclaration
+  ) {
+    requestUpdate.call(this, name, oldValue, options);
+    return requestUpdateThenable;
+  };
+}
+
+if (DEV_MODE) {
+  // Default warning set.
+  UpdatingElement.warnings = new Set(['change-in-update']);
```

Comment:
I'd probably make this field an array, just for the simpler syntax if someone wants to write `warnings = ['migrate']`. It's not going to be faster to search a set than a small list, and duplicates aren't a problem.

Reply:
Changed.